pkg/controller/history/controller_history.go

```diff
@@ -244,20 +247,25 @@ func (rh *realHistory) CreateControllerRevision(parent metav1.Object, revision *
 	}
 	clone := any.(*apps.ControllerRevision)
 
+	// First attempt to create the revision will not use collisionCount
```

Comment:
I'm not sure if the differences particularly the second one are intentional. IMO, the retry loop is better than deferring to the next sync cycle as name collision should be rare and when a collision does happen, it shouldn't take more than a couple of iterations to find a non-conflicting name. So the retry loop adds very little overhead. Again IMO, the code should be used generically by other controllers.